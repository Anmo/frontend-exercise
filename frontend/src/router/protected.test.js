import { render } from '@testing-library/react';
import { useLocation } from 'react-router-dom';
import { useRecoilValue } from 'recoil';

import ProtectedRoute from './protected';

jest.mock('react-router-dom', () => ({
	useLocation: jest.fn(),
	Redirect({ to: { pathname, state: { referrer }}}) {
		return <redirect {...{ pathname, referrer }} />
	},
	Route(props) {
		return <route {...props} />
	}
}));

jest.mock('recoil', () => ({
	...jest.requireActual('recoil'),
	useRecoilValue: jest.fn()
}));

test('renders the Redirect route (without userId', () => {
	useLocation.mockImplementation(() => ({
		pathname: 'test'
	}))

	const { container } = render(<ProtectedRoute />);

  expect(container.firstChild).toMatchSnapshot()
});

test('renders the normal Route (with userId', () => {
	useRecoilValue.mockImplementation(() => 'userId');
	useLocation.mockImplementation(() => ({}))

	const { container } = render(<ProtectedRoute path='/test'>Test</ProtectedRoute>);

  expect(container.firstChild).toMatchSnapshot()
});
