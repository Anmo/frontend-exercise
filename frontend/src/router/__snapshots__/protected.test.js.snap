// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Redirect route (without userId 1`] = `
<redirect
  pathname="/login"
  referrer="test"
/>
`;

exports[`renders the normal Route (with userId 1`] = `
<route
  path="/test"
>
  Test
</route>
`;
